### PR TITLE
feat: implement HelpText component

### PR DESCRIPTION
**Pull request by yvmunayev**
fix: #1607 

## Changes proposed in this PR:
- feat: implement ButtonHelp component

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/HelpText/index.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    /** A CSS class for the outer element, in addition to the component's base classes. */
+    className: PropTypes.string,
+    /** An object with custom style applied to the outer element. */
+    style: PropTypes.object,
+    /** Displayed the title of component. */
+    title: PropTypes.node,
+    /** Displayed the help message */
+    text: PropTypes.node.isRequired,
+    /** The icon to customize the button. If undefined it is used by default based on the variant. */
+    icon: PropTypes.node,
```
Comment (TahimiLeonBravo) [inline src/components/HelpText/index.js:109]: we are not add the`icon` prop right now

File: src/components/HelpText/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                ariaLabelledby={helpTextId}
+            >
+                {icon}
+            </StyledButton>
+            <InternalOverlay
+                isVisible={isOpen}
+                render={() => {
+                    return (
+                        <StyledTooltip id={helpTextId} role="tooltip">
+                            <RenderIf isTrue={!!title}>
```
Comment (TahimiLeonBravo) [inline src/components/HelpText/index.js:79]: The `title` always should be a `string` type, if the title is passed we render the text with the icon and if is not passed return null.

File: src/components/HelpText/styled/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import styled from 'styled-components';
+import attachThemeAttrs from '../../../styles/helpers/attachThemeAttrs';
+
+export const StyledButton = styled.button`
```
Comment (TahimiLeonBravo) [inline src/components/HelpText/styled/index.js:4]: I think we should use our `Button` component,
@reiniergs and @LeandroTorresSicilia whats do you think?

File: src/components/HelpText/styled/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    border: none;
+    background: transparent;
+
+    &:focus,
+    &:active {
+        outline: 0;
+    }
+`;
+export const StyledTooltip = attachThemeAttrs(styled.div)`
+    width: 244px;
```
Comment (TahimiLeonBravo) [inline src/components/HelpText/styled/index.js:14]: we should not add a fixed width, the user should control the width of the content they want to show, instead maybe we can use min-width

File: src/components/HelpText/styled/index.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    color: ${props => props.variantColor};
+    align-items: center;
+`;
+
+export const StyledIconContainer = styled.span`
+    margin-right: 8px;
+`;
+
+export const StyledText = attachThemeAttrs(styled.div)`
+    color: ${props => props.palette.text.main}
```
Comment (TahimiLeonBravo) [inline src/components/HelpText/styled/index.js:46]: also, we can add a `font-size` and `line-height`

File: src/components/HelpText/index.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+            event.preventDefault();
+            closeOverlay();
+        }
+    };
+
+    const icon = iconMap[variant] || iconMap.info;
+    const inverseIcon = inverseIconMap[variant] || inverseIconMap.info;
+
+    return (
+        <div id={id} className={className} style={style}>
```
Comment (TahimiLeonBravo) [inline src/components/HelpText/index.js:106]: I think we can't need to use a `div` wrapper here, instead we can use a Fragment `<>` and add the `className` `style` and `id` to the `StyledButton` element

File: src/components/HelpText/index.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    question: <QuestionInverseIcon />,
+    info: <InfoInverseIcon />,
+    error: <ErrorInverseIcon />,
+    warning: <WarningInverseIcon />,
+};
+
+/**
+ * HelpText is a popup that displays information related to an element.
+ */
+const HelpText = props => {
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/index.js:43]: to be consistent use function

**Pull request merged**